test/NuGet.Core.Tests/NuGet.Packaging.Test/LicensesTests/NuGetLicenseExpressionParserTests.cs

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root flicense information.
+
+using System;
+using System.Globalization;
+using Xunit;
+
+namespace NuGet.Packaging.Licenses.Test
+{
+    public class NuGetLicenseExpressionParserTests
+    {
+        [Theory]
+        [InlineData("MIT OR LPL-1.0", "OR", true)]
+        [InlineData("MIT AND LPL-1.0", "AND", true)]
+        [InlineData("MIT OR (AFL-1.1 WITH 389-exception)", "OR", true)]
+        [InlineData("(AFL-1.1 WITH 389-exception) OR MIT", "OR", true)]
+        [InlineData("MIT OR AFL-1.1 WITH 389-exception AND Apache-2.0", "OR", true)]
+        [InlineData("MIT OR AFL-1.1 WITH 389-exception", "OR", true)]
+        [InlineData("MIT OR (LPL-1.0 OR AFL-1.1 WITH 389-exception)", "OR", true)]
+        [InlineData("MIT OR (AFL-1.1 WITH 389-exception OR LPL-1.0)", "OR", true)]
+        [InlineData("MIT WITH 389-exception AND AFL-1.1 OR LPL-1.0", "OR", true)]
+        [InlineData("MIT OR AFL-1.1 AND LPL-1.0 WITH 389-exception", "OR", true)]
+        [InlineData("MIT OR ((AFL-1.1 WITH 389-exception) AND Apache-2.0)", "OR", true)]
+        [InlineData("((MIT) AND (LPL-1.0))", "AND", true)]
+        [InlineData("MIT OR (LPL-1.0 OR (AFL-1.1 WITH 389-exception))", "OR", true)]
+        [InlineData("MIT OR (LPL-1.0 AND (AFL-1.1 WITH 389-exception) AND AMDPLPA)", "OR", true)]
+        [InlineData("MIT OR ((AFL-1.1 WITH 389-exception) AND LPL-1.0)", "OR", true)]
+        [InlineData("MIT AND (AFL-1.1 WITH 389-exception OR Apache-2.0)", "AND", true)] // here. This expr is a weird legal expression, but a good test for the exp parser.
+        [InlineData("(AFL-1.1+ WITH 389-exception) OR MIT", "OR", true)]
+        [InlineData("MIT OR LPL-1.0+", "OR", true)]
+        [InlineData("(AMDPLPA AND BSD-2-Clause)", "AND", true)]
+        [InlineData("((( (AMDPLPA) AND BSD-2-Clause)))", "AND", true)]
+        [InlineData("(And+) AND or", "AND", false)]
```

Comment:
`AND AND AND` is valid but non-standard, right?

Reply:
Not quite. The license expression per SPDX is case sensitive. 
AND => operator
And => identifier, which will obviously get parsed as a non standard license. 

Per the design however, With WITH wITH would be invalid, as I did not find the concept of user defined exceptions.